less_users/tests/test_check_if_active.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+from django.urls import reverse
+from freezegun import freeze_time
+
+from datetime import date, timedelta
+
+from less_users.models import UserChallenge
+
+
+def test_check_if_active_before_duration_time(user, challenge_3_month):
+    with freeze_time(date.today() - timedelta(days=challenge_3_month.duration - 1)):
```

Comment:
To jest unittest funkcji, która oblicza wartość. Ta funkcja nie zapisuje nic do bazy. Bez zapisywania do bazy nie będziesz też potrzebowała freeze_time’a

Reply:
Teoretycznie rozumiem, że sprawdzam funkcję, ale ona jest powiązana z modelem, prawda? Więc zastanawiam się jak określę, że coś się dzieje najpierw w dniu X, a potem inne działanie następuje w dniu Z (czyli X + Y), skoro modele do bazy danych z automatu zapisują mi datę jako `models.DateTimeField(auto_now_add=True)` ?

Reply:
Wykorzystaj tego patcha i zobacz co się stanie. Pogadamy o Twoich wnioskach na spotkaniu
```git
diff --git a/less_users/tests/test_check_if_active.py b/less_users/tests/test_check_if_active.py
index 8edeefb..6ccc689 100644
--- a/less_users/tests/test_check_if_active.py
+++ b/less_users/tests/test_check_if_active.py
@@ -1,33 +1,32 @@
-import pytest
-from django.urls import reverse
 from freezegun import freeze_time
 
-from datetime import date, timedelta
+from datetime import date, timedelta, datetime
 
 from less_users.models import UserChallenge
 
 
 def test_check_if_active_before_duration_time(user, challenge_3_month):
-    with freeze_time(date.today() - timedelta(days=challenge_3_month.duration - 1)):
-        userchallenge = UserChallenge.objects.create(
-            user=user, challenge=challenge_3_month
-        )
+    start_date = datetime.utcnow() - timedelta(days=challenge_3_month.duration - 1)
+    userchallenge = UserChallenge(
+        user=user, challenge=challenge_3_month, start_date=start_date)
 
     assert userchallenge.check_if_active
+    assert userchallenge.start_date == start_date
 
 
 def test_check_if_active_after_duration_time(user, challenge_3_month):
-    with freeze_time(date.today() - timedelta(days=challenge_3_month.duration + 1)):
-        userchallenge = UserChallenge.objects.create(
-            user=user, challenge=challenge_3_month
-        )
+    start_date = datetime.utcnow() - timedelta(days=challenge_3_month.duration + 1)
+    userchallenge = UserChallenge(
+        user=user, challenge=challenge_3_month, start_date=start_date
+    )
 
     assert not userchallenge.check_if_active
+    assert userchallenge.start_date == start_date
 
 
 def test_check_if_not_active_when_deactivated(user, challenge_3_month):
-    userchallenge = UserChallenge.objects.create(
-        user=user, challenge=challenge_3_month, is_active=False
+    userchallenge = UserChallenge(
+        user=user, challenge=challenge_3_month, is_active=False, start_date=datetime.utcnow()
     )
 
     assert not userchallenge.check_if_active
```

Reply:
Mira, dziękuję! 
1. już nie tylko wiem jak zrobić **patcha** z aktualnego statusu kodu, ale też jak go **apply** do kodu (bez przepisywania zmian ;) )
2. nie wiem czemu, ale pomyślałam, że jak mam w modelu ustawione `auto_now_add`, to pole jest "zablokowane" przed podaniem mu własnej wartości... 
... ale Twoja propozycja działa bez zarzutu, bez dodawania freeze_gun, czyli znów nie pomyślałam, a można było prościej.

Reply:
ale oczywiście możemy to omówić wieczorem :)